### PR TITLE
test: Migrate Backend tests to a GHA workflow

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**
Resolves https://github.com/kubeflow/pipelines/issues/10867

Converts the Prow backend test to a GH Action Workflow.

Here's a working GH Action run on my fork: https://github.com/DharmitD/data-science-pipelines-argo/actions/runs/9390063669

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: .github/workflows/presubmit-backend.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    paths:
+      - 'backend/**'
+      - 'test/presubmit-backend-test.sh'
+  workflow_dispatch:
+
```
Comment (hbelmiro) [inline .github/workflows/presubmit-backend.yml:14]: You can remove these lines.

**Review by hbelmiro — COMMENTED** [review]
In parallel to this PR, we need to remove `kubeflow-pipeline-e2e-test` from [GoogleCloudPlatform/oss-test-infra](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/3738b70f78b971cc281b1791cf8e73cfbe08ae55/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L15).

File: .github/workflows/presubmit-backend.yml (hunk 1 above)
Comment (DharmitD) [reply .github/workflows/presubmit-backend.yml]: Done.

**Review by hbelmiro — COMMENTED** [review]
/lgtm

**Comment by HumairAK** [comment]
the tests in github actions look like they are running great: 

Github workflow: https://github.com/kubeflow/pipelines/actions/runs/9390915591/job/25861839671?pr=10871
From Prow: https://oss.gprow.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/10871/kubeflow-pipeline-backend-test/1798454199197896704

/lgtm
/approve

@chensun

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thank you @DharmitD !

**Comment by DharmitD** [comment]
> In parallel to this PR, we need to remove `kubeflow-pipeline-e2e-test` from [GoogleCloudPlatform/oss-test-infra](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/3738b70f78b971cc281b1791cf8e73cfbe08ae55/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L15).

Proposed a PR to remove this test from Prow: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2296

**Pull request merged**